src/rpc/util.cpp

```diff
@@ -40,20 +40,15 @@ void RPCTypeCheck(const UniValue& params,
             break;
 
         const UniValue& v = params[i];
-        if (!(fAllowNull && v.isNull())) {
-            RPCTypeCheckArgument(v, t);
+        if (fAllowNull && v.isNull()) continue;
+        if (!t.typeAny && v.type() != t.type) {
+            throw JSONRPCError(RPC_TYPE_ERROR,
+                               strprintf("JSON value of type %s is not of expected type %s", uvTypeName(v.type()), uvTypeName(t.type)));
```

Comment:
Would be nice to split this behaviour change of unifying the string into a separate pull request from the refactoring changes

Reply:
ok, one sec.